org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/util/Util.java

```diff
@@ -1001,6 +1003,9 @@ public static IPackageFragment getPackageFragment(char[] fileName, int pkgEnd, i
 		if (jarSeparator != -1) {
 			String jarMemento = new String(fileName, 0, jarSeparator);
 			PackageFragmentRoot root = (PackageFragmentRoot) JavaCore.create(jarMemento);
+			if( root == null ) {
+				root = getJarPkgFragmentRoot(new String(fileName), jarSeparator, jarMemento);
+			}
```

Comment:
what motivates this change in a PR addressing a problem in ClassFileWorkingCopy?

Reply:
The tests continued to fail will the following stacktrace. I realize this will be impossible to actually follow, as you don't have our underlying implementations. I could try to break this part out and come up with a new specific test.  But anyway, here goes: 

```
Util.getPackageFragment(char[], int, int) line: 1011	
Util.getClassFile(char[]) line: 810	
Util.getUnresolvedJavaElement(TypeBinding, WorkingCopyOwner, Util$BindingsToNodesMap) line: 1611	
TypeBinding.getUnresolvedJavaElement(TypeBinding) line: 566	
TypeBinding.getUnresolvedJavaElement() line: 557	
TypeBinding.getJavaElement() line: 539	
```

`getPackageFragment(char[] fileName, int pkgEnd, int jarSeparator)` appears to be set up to handle inside-jar paths, but, for some reason, it wasn't working for me. In this case, the values passed in were:

```
fileName=/home/rob/apps/eclipse/workspaces/jclMin1.8.jar|java/lang/RuntimeException.class
pkgEnd=57
jarSeparator=47
```

The call to `PackageFragmentRoot root = (PackageFragmentRoot) JavaCore.create(jarMemento);` was returning null  when `jarMemento` has a value of `/home/rob/apps/eclipse/workspaces/jclMin1.8.jar`

I'm not familiar enough with this code to know why it's doing things this way. It appears to me that the currently implementation simply did not function for this specific case. But I did know i had browsed similar code that did properly turn a path of this structure into a PackageFragmentRoot without error, and so I copied it over, and it worked. 

I still maintain that this change to locating the PackageFragmentRoot is at the very least closely related to changing the `getFileName()` implementation from returning the path of the jar to the full path of the resource inside the jar. If you prefer, I can try to break this specific issue out and try to develop a test case for it.